tests/test_ssc.py

```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/python
+import numpy as np
+import os
+import pytest
+import tjpcov.main as cv
+from tjpcov.parser import parse
+import yaml
+import sacc
+import glob
+import pyccl as ccl
+import healpy as hp
+import shutil
+
+
+root = "./tests/benchmarks/32_DES_tjpcov_bm/"
+input_yml_ssc = os.path.join(root, "tjpcov_conf_minimal_ssc.yaml")
+
+
+def clean_tmp():
+    if os.path.isdir('./tests/tmp'):
+        shutil.rmtree('./tests/tmp/')
+    os.makedirs('./tests/tmp')
+
+    if os.path.isdir('./tests/benchmarks/32_DES_tjpcov_bm/tjpcov_tmp/'):
+        shutil.rmtree('./tests/benchmarks/32_DES_tjpcov_bm/tjpcov_tmp/')
+    os.makedirs('./tests/benchmarks/32_DES_tjpcov_bm/tjpcov_tmp/')
+
+clean_tmp()
+
+def get_config():
+    with open(input_yml_ssc) as f:
+        config = yaml.safe_load(f)
+    return config
+
+
+def get_halomodel_calculator_and_NFW_profile(cosmo):
+    md = ccl.halos.MassDef200m()
+    mf = ccl.halos.MassFuncTinker08(cosmo, mass_def=md)
+    hb = ccl.halos.HaloBiasTinker10(cosmo, mass_def=md)
+    cm = ccl.halos.ConcentrationDuffy08(mdef=md)
+
+    hmc = ccl.halos.HMCalculator(cosmo, mf, hb, md)
+    pNFW = ccl.halos.HaloProfileNFW(cm)
+
+    return hmc, pNFW
+
+
+def get_cl_footprint(tr1, tr2, tr3, tr4):
+    config = get_config()
+    mf = config['tjpcov']['mask_file']
+    mf12 = hp.read_map(mf[tr1]) * hp.read_map(mf[tr2])
+    mf34 = hp.read_map(mf[tr3]) * hp.read_map(mf[tr4])
+
+    alm = hp.map2alm(mf12)
+    blm = hp.map2alm(mf34)
+
+    cl = hp.alm2cl(alm * blm)
+    cl *= (2 * np.arange(cl.size) + 1)
+
+    return cl
+
+
+def get_CovarianceCalculator():
+    config = get_config()
+    return cv.CovarianceCalculator(config)
+
+
+@pytest.mark.parametrize('tracer_comb1,tracer_comb2',
+                         [(('DESgc__0', 'DESgc__0'), ('DESgc__0', 'DESgc__0')),
+                          (('DESgc__0', 'DESwl__0'), ('DESwl__0', 'DESwl__0')),
+                          (('DESgc__0', 'DESgc__0'), ('DESwl__0', 'DESwl__0')),
+                          (('DESwl__0', 'DESwl__0'), ('DESwl__0', 'DESwl__0')),
+                          (('DESwl__0', 'DESwl__0'), ('DESwl__1', 'DESwl__1')),
+                          ])
+def test_get_SSC_cov(tracer_comb1, tracer_comb2):
+    # TJPCov covariance
+    cc  = get_CovarianceCalculator()
+    cosmo = cc.cosmo
+    s = cc.cl_data
+    ell, _ = s.get_ell_cl('cl_00', 'DESgc__0', 'DESgc__0')
+
+    ccl_tracers, _ = cc.get_tracer_info(s)
+    cov_ssc = cc.get_SSC_cov(tracer_comb1=tracer_comb1,
+                             tracer_comb2=tracer_comb2,
+                             ccl_tracers=ccl_tracers,
+                             integration_method='qag_quad',
+                             include_b_modes=False)
+
+    # Check saved file
+    covf = np.load(cc.outdir + '/ssc_{}_{}_{}_{}.npz'.format(*tracer_comb1,
+                                                             *tracer_comb2))
+    assert np.max(np.abs((covf['cov_nob'] + 1e-100) / (cov_ssc + 1e-100) - 1)) < 1e-10
```

Comment:
No, it's because there will be terms that are exactly 0 (in both numerator and denominator) and then you will get nan's if you divide by them.

Reply:
Is np.allclose or np.testing.assert_allclose suitable for this case?

Reply:
yes, it would work. I kind of prefer the `np.max(np.abs())` way because when it fails it's easier to see in the printed message for how much it fails.